lib/migrate.js

```diff
@@ -42,6 +47,30 @@ function migrate (set, direction, migrationName, fn) {
         next(migrations.shift())
       })
     })
+
+    if (migrationPromise && migrationPromise.then) {
+      migrationPromise
+        .then(function () {
+          if (migrationCompleted) {
+            return fn(new Error('Migration was already run. Please provide callback OR promise, not both.'))
+          }
+
+          migrationCompleted = true
+          // Set timestamp if running up, clear it if down
+          migration.timestamp = direction === 'up' ? Date.now() : null
+
+          // Decrement last run index
+          lastRunIndex--
+
+          set.lastRun = direction === 'up' ? migration.title : set.migrations[lastRunIndex] && set.migrations[lastRunIndex].title
+          set.save(function (err) {
+            if (err) return fn(err)
+
+            next(migrations.shift())
+          })
```

Comment:
Is this fine, or should i split the callback out into a separate function, which can be reused in both callback & promise cases?

Reply:
Yeah, I would do that for sure.